pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/SignatureDeclareThrowsExceptionRule.java

```diff
@@ -39,172 +34,78 @@
+            addViolation(o, methodDeclaration);
         }
 
-        // Ignore overridden methods, the issue should be marked on the method definition
-        final List<ASTAnnotation> methodAnnotations = methodDeclaration.getParent().findChildrenOfType(ASTAnnotation.class);
-        for (final ASTAnnotation annotation : methodAnnotations) {
-            final ASTName annotationName = annotation.getFirstDescendantOfType(ASTName.class);
-            if (annotationName.hasImageEqualTo("Override") || annotationName.hasImageEqualTo("java.lang.Override")) {
-                return super.visit(methodDeclaration, o);
-            }
-        }
-
-        checkExceptions(methodDeclaration, o);
-
         return super.visit(methodDeclaration, o);
     }
 
-    private boolean isAllowedMethod(ASTMethodDeclaration methodDeclaration) {
-        if (getProperty(IGNORE_JUNIT_COMPLETELY_DESCRIPTOR)) {
-            return true;
-        } else {
-            return "setUp".equals(methodDeclaration.getName())
-                    || "tearDown".equals(methodDeclaration.getName());
-        }
-    }
-
     @Override
     public Object visit(ASTConstructorDeclaration constructorDeclaration, Object o) {
-        if (junitImported && getProperty(IGNORE_JUNIT_COMPLETELY_DESCRIPTOR)) {
+        if (junit3TestClass && getProperty(IGNORE_JUNIT_COMPLETELY_DESCRIPTOR)) {
             return super.visit(constructorDeclaration, o);
         }
 
-        checkExceptions(constructorDeclaration, o);
-        return super.visit(constructorDeclaration, o);
-    }
-
-    /**
-     * Search the list of thrown exceptions for Exception
-     */
-    private void checkExceptions(Node method, Object o) {
-        List<ASTName> exceptionList = Collections.emptyList();
-        ASTNameList nameList = method.getFirstChildOfType(ASTNameList.class);
-        if (nameList != null) {
-            exceptionList = nameList.findDescendantsOfType(ASTName.class);
-        }
-        if (!exceptionList.isEmpty()) {
-            evaluateExceptions(exceptionList, o);
+        if (checkExceptions(constructorDeclaration.getGenericSignature())) {
+            addViolation(o, constructorDeclaration);
         }
+        return super.visit(constructorDeclaration, o);
     }
 
     /**
      * Checks all exceptions for possible violation on the exception
      * declaration.
      *
-     * @param exceptionList
-     *            containing all exception for declaration
-     * @param context
+     * @return true if "java.lang.Exception" has been declared
      */
-    private void evaluateExceptions(List<ASTName> exceptionList, Object context) {
-        for (ASTName exception : exceptionList) {
-            if (hasDeclaredExceptionInSignature(exception)) {
-                addViolation(context, exception);
+    private boolean checkExceptions(JMethodSig signature) {
+        for (JTypeMirror exception : signature.getThrownExceptions()) {
+            if (TypeTestUtil.isA(Exception.class, exception)) {
```

Comment:
This will test subtypes of Exception...

You could make this function shorter:
```java
return CollectionUtil.any(signature.getThrownExceptions(), e -> TypeTestUtil.isExactlyA(Exception.class, e));
```